packages/fx-core/src/plugins/resource/apiconnector/questions.ts

```diff
@@ -38,6 +41,57 @@ export class BaseQuestionService {
     this.logger = logger;
   }
 }
+
+export class ComponentsQuestion extends BaseQuestionService implements IQuestionService {
+  protected readonly ctx: Context;
+  protected readonly components: OptionItem[];
+  protected readonly projectPath: string;
+  constructor(
+    ctx: Context,
+    componentOptions: OptionItem[],
+    projectPath: string,
+    telemetryReporter?: TelemetryReporter,
+    logger?: LogProvider
+  ) {
+    super(telemetryReporter, logger);
+    this.ctx = ctx;
+    this.components = componentOptions;
+    this.projectPath = projectPath;
+  }
+  public getQuestion(): MultiSelectQuestion {
+    return {
+      type: "multiSelect",
+      name: Constants.questionKey.componentsSelect,
+      title: getLocalizedString("plugins.apiConnector.whichService.title"),
+      staticOptions: this.components,
+      onDidChangeSelection: async (currentSelectedIds: Set<string>): Promise<Set<string>> => {
+        if (!this.ctx || !this.projectPath) {
```

Comment:
when will this scenario happen?

Reply:
not sure, since CLI scenario has no ctx and projectPath